src/commands/schema/push.js

```diff
@@ -0,0 +1,127 @@
+const FaunaCommand = require("../../lib/fauna-command.js");
+const fetch = require("node-fetch");
+const fs = require("fs");
+const path = require("path");
+const { Flags, ux } = require("@oclif/core");
+const FormData = require("form-data");
+
+const FILE_LIMIT = 256;
+const FILESIZE_LIMIT_BYTES = 32 * 1024 * 1024;
+
+class PushSchemaCommand extends FaunaCommand {
+  async run() {
+    const {
+      connectionOptions: { domain, port, scheme, secret },
+    } = await this.getClient();
+    const dir = this.flags.dir;
+    const filenames = fs.readdirSync(dir);
+    var totalsize = 0;
+    const files = filenames
+      .filter(
+        (filename) =>
+          filename.endsWith(".fsl") &&
+          !fs.statSync(path.join(dir, filename)).isDirectory()
+      )
+      .map((filename) => {
+        const content = fs.readFileSync(path.join(dir, filename));
+        totalsize += content.length;
+        if (totalsize > FILESIZE_LIMIT_BYTES) {
+          this.error(
+            `Too many bytes: at most ${FILESIZE_LIMIT_BYTES} may be pushed`
+          );
+        }
+        return {
+          name: filename,
+          content: content,
+        };
+      });
+    if (files.length > FILE_LIMIT) {
+      this.error(`Too many files: ${files.length} > ${FILE_LIMIT}`);
+    }
+    const body = () => {
+      const fd = new FormData();
+      for (const file of files) {
+        fd.append(file.name, Buffer.from(file.content));
+      }
+      return fd;
+    };
+
+    try {
+      if (this.flags.force) {
+        // Just push.
+        const res = await fetch(
+          `${scheme}://${domain}:${port}/schema/1/update?force=true`,
+          {
+            method: "POST",
+            headers: { AUTHORIZATION: `Bearer ${secret}` },
+            body: body(),
+          }
+        );
+        const json = await res.json();
+        if (json.error) {
+          this.error(json.error.message);
+        }
+      } else {
+        // Confirm diff, then push it.
+        var vurl = `${scheme}://${domain}:${port}/schema/1/validate?force=true`;
```

Comment:
remind me, what does `?force` do on `/validate`?

also this should use `let` not `var`. var declares a function-global variable in JS, whereas `let` has the same scoping mechanics as `const`

Reply:
Even validate wants a version to do a version check, so `force` skips the check so we can get back a version to use for push.